ui/app/utils/properties/job-client-status.js

```diff
@@ -26,7 +26,10 @@ export default function jobClientStatus(nodesKey, jobKey) {
 
       // Filter nodes by the datacenters defined in the job.
       const filteredNodes = nodes.filter((n) => {
-        return job.datacenters.indexOf(n.datacenter) >= 0;
+        return (
+          job.datacenters.includes('*') ||
```

Comment:
`*` in datacenters is not like namespace, where `*` is a special wildcard to match all. In datacenters it's actually a glob pattern so, for example, `datacenters = ["dc*"]` would match `dc1` and `dc2` but not `us-east-1`. This check would return all clients, even the ones not matched by the pattern

Reply:
Ahh thanks for keeping an eye out. I'll update to match on glob.

Reply:
Returning all clients here could unnecessarily throw a `403`.

Reply:
+1 to @lgfa29 -- you probably want to extract the logic from `_doesMatchPattern` into a utility and use it here.

Reply:
Yup, that looks pretty good! If you need some test cases to match with the Go implementation:
https://github.com/ryanuber/go-glob/blob/master/glob_test.go